### PR TITLE
Add support for Torch's Conv1d strides and ConvTranspose1d

### PR DESCRIPTION
**Pull request by fcaspe**
Hi Jatin!

This library is really awesome, I have been using it for low latency inference of big convolutional autoencoders, so I have implemented the 1d Transposed Convolution and convolutional strides.

 - ConvTranspose1d is implemented with RTNeural's `Conv1D` class, but a different loading function has to be called, `RTNeural::torch_helpers::loadConvTranspose1D`. See `torch_convtranspose1d_test.cpp` for an example.

 - Conv1d strides are implemented using a `.skip()` method that performs a single stride step. This just updates the circular buffer of the `Conv1D` layer with the new input we jump over. For example, if strides=2 is required, then `.skip()` has to be called every time after a `.forward()` call is made. See `torch_conv1d_stride_test.cpp` for an example.
 
I know these new functionalities are not fully incorporated into the library. For instance, strides are still missing in `Conv1DT` and the non-streaming versions of `Conv1D` and `Conv2D`. Let me know what you think about these additions and I will be happy to improve them so that hopefully they can be integrated into the library!

Best,
Franco

### CODE REVIEW

File: tests/functional/torch_convtranspose1d_test.cpp [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+{
+    testTorchConvTranspose1DModel<float,4,15,5,3,1,1,3>(
+        "models/convtranspose1d_torch.json",
+    "test_data/convtranspose1d_torch_x_python.csv",
+    "test_data/convtranspose1d_torch_y_python.csv");
+}
+
+TEST(TestTorchConvTranspose1D, modelOutputMatchesPythonImplementationForDoubles)
+{
+    testTorchConvTranspose1DModel<float,4,15,5,3,1,1,3>(
```
Comment (jatinchowdhury18) [inline tests/functional/torch_convtranspose1d_test.cpp:135]:
```suggestion
    testTorchConvTranspose1DModel<double,4,15,5,3,1,1,3>(
```

File: tests/functional/torch_convtranspose1d_test.cpp [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+{
+    testStreamingTorchConvTranspose1DModel<float,4,15,5,3,1,1,3>(
+        "models/convtranspose1d_torch.json",
+    "test_data/convtranspose1d_torch_x_python_cc.csv",
+    "test_data/convtranspose1d_torch_y_python_cc.csv");
+}
+
+TEST(TestTorchConvTranspose1D, streaming_modelOutputMatchesPythonImplementationForDoubles)
+{
+    testStreamingTorchConvTranspose1DModel<float,4,15,5,3,1,1,3>(
```
Comment (jatinchowdhury18) [inline tests/functional/torch_convtranspose1d_test.cpp:152]:
```suggestion
    testStreamingTorchConvTranspose1DModel<double,4,15,5,3,1,1,3>(
```

**Review by jatinchowdhury18 — COMMENTED** [review]
Hello! These changes look great.

I think the "ConvTranspose" changes are good to go. I guess it would be cool to add some tests to double-check that it works correctly with the compile-time API as well, but  I guess that's not 100% necessary.

For the Conv1D strides, the "skip" implementation looks correct. I would love to take a shot at implementing that in the compile-time implementations of the Conv1D layer as well... I have a rough idea how that should work. I'm also thinking about making a "wrapper" with a counter to keep track of whether `forward()` or `skip()` should be called.

Do you mind if I push any changes back to your branch?

**Comment by jatinchowdhury18** [comment]
Alright, I think I've made all the changes that I want to make... Still need to do a pass for cleanup and documentation, but if @fcaspe wants to have another look and make sure I didn't mess things up too bad, that would be much appreciated :).

**Comment by fcaspe** [comment]
Just reviewed. The 'StridedConv1d` class is a good idea. The examples look good and cleaner than mine. I also tried on the conv models I am developing and they are working ok with this new version!

**Comment by jatinchowdhury18** [comment]
@fcaspe Awesome! I'm going to go ahead and merge this PR.

**Pull request merged**